docs/cli.md

```diff
@@ -218,6 +219,19 @@ Usage:
 $ s2i rebuild <image name> [<new-tag-name>]
 ```
 
+# s2i generate
+The `s2i generate` command produces a Dockerfile using an existing S2I base 
+image that can be used to produce an image by any application supporting the
+format.
```

Comment:
I would add a "tech preview" note:

```
This command is a Technology Preview feature, and is subject to change or be removed without notice in a future release.
```

The CLI usage should also be updated. In general we try to keep the CLI help docs and the text here identical.

Reply:
Should the CLI usage include the "tech preview" note as well?

Reply:
Done in https://github.com/openshift/source-to-image/pull/1023/commits/a9c3f7ff062161628a57f1c5948b09a4584e1f5d